File: Program/python_and_ruby_101/OOP_Variable/3.py
class Cal(object) :
    #생성자
    def __init__(self,v1,v2):
        #첫번째 매개변수 반드시 지정해야하고
        #이름은 상관없지만 뭐든 지정해야 instance로 알아들음
        if isinstance(v1, int):
            self.v1 = v
        if isinstance(v2, int):
            self.v2 = v
    def add(self):
        return self.v1 + self.v2
    def subtract(self):
        return self.v1 - self.v2
    def setV1(self,v):
        if isinstance(v, int):
            self.v1 = v
    def getV1(self) :
        return self.v1

c1 = Cal(10,10)
print(c1.add())
c1.v1 =20
c1.v2 =30
print(c1.add())
print(c1.subtract())

File: Program/python_and_ruby_101/OOP_Variable/3.rb
class Cal
  def initialize(v1,v2)
    #생성자 : 인스턴스 만들어질 때 반드시 실행되어야 하는 내용
    p v1, v2
    @v1 = v1
    @v2 = v2
  end
  def add()
    return @v1+@v2
  end
  def subtract()
    return @v1-@v2
  end
  def getValue()
    return @value
  end
  def setV1(v)
    if v.is_a?(Integer)
      @v1 = v
    end
  end
  def getV1()
    return @v1
  end
end

c1 = Cal.new(10,10)
p(c1.add())
p(c1.subtract())
c1.setV1('하이')
c1.add()
c2 = Cal.new(30,20)
p(c2.add())
p(c2.subtract())
